internal/host/transition.go

```diff
@@ -544,6 +540,14 @@ func (th *transitionHandler) IsDay2Host(sw stateswitch.StateSwitch, args statesw
 	return hostutil.IsDay2Host(sHost.host), nil
 }
 
+func (th *transitionHandler) HostNotResponsiveWhileInstallation(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
+	sHost, ok := sw.(*stateHost)
+	if !ok {
+		return false, errors.New("HasClusterError incompatible type of StateSwitch")
+	}
+	return funk.Contains(disconnectionValidationStages, sHost.host.Progress.CurrentStage) && !hostIsResponsive(sHost.host), nil
```

Comment:
It does not cover MGMT-3299, when host is in `HostStatusInstalling` state it does not have any progress stage.
I think that you need to split the transition to handle `HostStatusInstalling` and `HostStatusInstallingInProgress` separately